dimagi/commcare-hq
# Upgrade Stripe

## Technical Summary

Ticket: https://dimagi-dev.atlassian.net/browse/SAAS-15095
Upgrade Stripe SDK from `4.1.0` to `6.7.0` , review change logs for SDK
Upgrade Stripe API from `2011-11-17` to `2023-08-16` , review change logs for API

## Safety Assurance

### Safety story

Tested locally. Tests cover every stripe api calls are all passed. Will have QA test it as well.

### Automated test coverage


`test_autopay_with_api`, `test_stripe_payment_method_with_api`, `test_stripe_utils_with_api`

### QA Plan


QA Ticket: https://dimagi-dev.atlassian.net/browse/QA-5826

Because we upgrade Stripe, QA should test all places that is related to payment, billing record, autopay, save card, unsave card, ...


### Rollback instructions




### Labels & Review
